File: java/03-bootstrap/src/test/java/com/fResult/bootstrap/customers/services/DataSourceCustomerServiceTest.java
package com.fResult.bootstrap.customers.services;

import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseBuilder;
import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseType;

class DataSourceCustomerServiceTest extends BaseClass {
  private final CustomerService customerService;

  DataSourceCustomerServiceTest() {
    final var dataSource =
        new EmbeddedDatabaseBuilder()
            .setType(EmbeddedDatabaseType.H2)
            .addScript("classpath:schema.sql")
            .build();

    customerService = new DataSourceCustomerService(dataSource);
  }

  @Override
  public CustomerService getCustomerService() {
    return customerService;
  }
}

File: kotlin/03-bootstrap/src/test/kotlin/com/fResult/bootstrap/customers/services/DataSourceCustomerServiceTest.kt
package com.fResult.bootstrap.customers.services

import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseBuilder
import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseType

class DataSourceCustomerServiceTest() : BaseClass() {
  private val customerService: CustomerService

  init {
    EmbeddedDatabaseBuilder()
      .setType(EmbeddedDatabaseType.H2)
      .addScript("classpath:schema.sql")
      .build()
      .also { customerService = DataSourceCustomerService(it) }
  }

  override fun getCustomerService(): CustomerService {
    return customerService
  }
}
